spec/Section 5 -- Validation.md

```diff
@@ -604,9 +604,9 @@ fragment scalarSelectionsNotAllowedOnInt on Dog {
 }
 ```
 
-Conversely the leaf field selections of GraphQL operations must be of type
-scalar or enum. Leaf selections on objects, interfaces, and unions without
-subfields are disallowed.
+Conversely the underlying type of leaf field selections of GraphQL operations
+must be scalar or enum. Leaf selections on fields whose underlying type is
+object, interface, or union are disallowed.
```

Comment:
"Leaf selections ... without subfields" seemed redundant to me - a leaf selection is one without subfields, right?

Reply:
Aren't leaf selections always fields? Making this whole section about "leaf fields" instead of "leaf selections" might clear up the language a bit?

Reply:
Ah 🤦‍♂️  I shouldn't raise PRs before I've had breakfast. I've re-added "without subfields" for now, we could do a more significant rewrite via a later PR.